packages/tooltip/src/index.tsx

```diff
@@ -285,6 +292,55 @@ function useTooltip<T extends HTMLElement>({
     return () => ownerDocument.removeEventListener("keydown", listener);
   }, []);
 
+  React.useEffect(() => {
+    /*
+      This is a workaround for using tooltips with disabled controls in Safari.
+      Safari fires `pointerenter` but does not fire `pointerleave`
+      and `onPointerEventLeave` added to the trigger element will not work
+    */
+    if (!("PointerEvent" in window) || !disabled) return;
+
+    let ownerDocument = getOwnerDocument(ownRef.current)!;
+
+    function listener(event: MouseEvent) {
+      if (state !== VISIBLE) return;
+
+      if (
+        event.target instanceof Element &&
+        event.target.closest(
+          "[data-reach-tooltip-trigger][data-state='tooltip-visible']"
+        )
+      ) {
+        return;
+      }
+
+      transition(GLOBAL_MOUSE_MOVE);
+    }
+
+    ownerDocument.addEventListener("mousemove", listener);
+    return () => ownerDocument.removeEventListener("mousemove", listener);
+  }, [disabled]);
+
+  function wrapMouseEvent<EventType extends React.SyntheticEvent | Event>(
+    theirHandler: ((event: EventType) => any) | undefined,
+    ourHandler: (event: EventType) => any
+  ) {
+    // Use internal MouseEvent handler only if PointerEvent not supported
+    if ("PointerEvent" in window) return theirHandler;
```

Comment:
Hmm... seems like this line is causing Gatsby to fail on server-side rendering:

```
$ cross-env NODE_ENV=production gatsby build
failed Building static HTML for pages - 0.750s
error "window" is not available during server side rendering.

  246 |   function wrapMouseEvent(theirHandler, ourHandler) {
  247 |     // Use internal MouseEvent handler only if PointerEvent is not supported
> 248 |     if ("PointerEvent" in window) return theirHandler;
      | ^
  249 |     return wrapEvent(theirHandler, ourHandler);
  250 |   }
  251 | 


  WebpackError: ReferenceError: window is not defined
  
  - tooltip.esm.js:248 
    /[@reach]/tooltip/dist/tooltip.esm.js:248:1
  
  - tooltip.esm.js:333 
    /[@reach]/tooltip/dist/tooltip.esm.js:333:1
  

not finished Generating image thumbnails - 147.857s
```

Reported in https://github.com/reach/reach-ui/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc